python/taichi/main.py

```diff
@@ -98,6 +98,9 @@ def get_available_examples() -> set:
 def run_example(name: str):
     """Run an example based on the example NAME."""
     all_example_names = get_available_examples()
+    # try to find and remove python file extension once
+    if name.endswith('.py'):
+        name = name.split('.')[0]
```

Comment:
```suggestion
        name = name[:-3]
```
In case of `example.name.py` -> `example`.

Reply:
Thanks for the comment! I think I will refactor this PR once #1085 is merged(as it requires to rebase)